src/dash-playlist-loader.js

```diff
@@ -696,45 +710,63 @@ export default class DashPlaylistLoader extends EventTarget {
-    // if the minimumUpdatePeriod was changed, update the minimumUpdatePeriodTimeout_
-    if (!oldMaster || (newMaster && oldMaster.minimumUpdatePeriod !== newMaster.minimumUpdatePeriod)) {
+    if (!oldMaster || (newMaster && newMaster.minimumUpdatePeriod !== oldMaster.minimumUpdatePeriod)) {
       this.updateMinimumUpdatePeriodTimeout_();
     }
 
     return Boolean(newMaster);
   }
 
   updateMinimumUpdatePeriodTimeout_() {
-    // Clear existing timeout
-    window.clearTimeout(this.minimumUpdatePeriodTimeout_);
+    const mpl = this.masterPlaylistLoader_;
 
-    const createMUPTimeout = (mup) => {
-      this.minimumUpdatePeriodTimeout_ = window.setTimeout(() => {
-        this.trigger('minimumUpdatePeriod');
-        createMUPTimeout(mup);
-      }, mup);
-    };
+    // cancel any pending creation of mup on media
+    // a new one will be added if needed.
+    if (mpl.createMupOnMedia_) {
+      mpl.off('loadedmetadata', mpl.createMupOnMedia_);
+      mpl.createMupOnMedia_ = null;
+    }
 
-    const minimumUpdatePeriod = this.masterPlaylistLoader_.master && this.masterPlaylistLoader_.master.minimumUpdatePeriod;
+    // clear any pending timeouts, master was just updated
+    // so if minimumUpdatePeriodTimeout_ is still valid
+    window.clearTimeout(mpl.minimumUpdatePeriodTimeout_);
+    mpl.minimumUpdatePeriodTimeout_ = null;
+    mpl.createMUPTimeout_();
+  }
 
-    if (minimumUpdatePeriod > 0) {
-      createMUPTimeout(minimumUpdatePeriod);
+  createMUPTimeout_() {
+    const mpl = this.masterPlaylistLoader_;
+    let mup = mpl.master && mpl.master.minimumUpdatePeriod;
 
     // If the minimumUpdatePeriod has a value of 0, that indicates that the current
     // MPD has no future validity, so a new one will need to be acquired when new
     // media segments are to be made available. Thus, we use the target duration
     // in this case
-    } else if (minimumUpdatePeriod === 0) {
-      // If we haven't yet selected a playlist, wait until then so we know the
-      // target duration
-      if (!this.media()) {
-        this.one('loadedplaylist', () => {
-          createMUPTimeout(this.media().targetDuration * 1000);
-        });
+    if (mup === 0) {
+      if (mpl.media()) {
+        mup = mpl.media().targetDuration * 1000;
       } else {
-        createMUPTimeout(this.media().targetDuration * 1000);
+        mpl.createMupOnMedia_ = mpl.updateMinimumUpdatePeriodTimeout_;
+        mpl.one('loadedmetadata', mpl.createMupOnMedia_);
       }
     }
+
+    // if minimumUpdatePeriod is invalid or <= zero skip creating a timeout
+    if (typeof mup !== 'number' || mup <= 0) {
+      return;
+    }
+
+    mpl.minimumUpdatePeriodTimeout_ = window.setTimeout(() => {
+      mpl.minimumUpdatePeriodTimeout_ = null;
+      mpl.createMUPTimeout_();
+
+      // if a minimumUpdatePeriodTimeout_ was created again
+      // then the minimumUpdatePeriod was valid
+      // trigger minimumUpdatePeriod
+      if (mpl.minimumUpdatePeriodTimeout_) {
```

Comment:
It looks like before this change, `minimumUpdatePeriod` was triggered regardless of whether a new one was created. Any specific reason for this change?